sinks/sinks.go

```diff
@@ -35,6 +41,7 @@ const EventReportedCount = "sink.events_reported_total"
 // it's aggregated metrics.
 type MetricSink interface {
 	Name() string
+	Kind() string
```

Comment:
Also, an idea to reduce a bit of boilerplate and codemod toil: could we wrap all of the sinks in a wrapper struct that can handle `Name()` and `Kind()` so each sink doesn't have to implement it? It might be weird / unergonomic though given a lack of proper generics support in golang.

Reply:
I like this idea, but want to punt on it for a future PR.